### PR TITLE
FocusTrapZone: Add/remove focus and click handlers when props change …

### PR DESCRIPTION
**Pull request by ecraig12345**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes

This is a port of #6717 to Fabric 5 (description pasted below). Also includes @natalieethell's fix for #5704, so that panels which set `isHiddenOnDismiss` to true will not continue to trap focus while hidden.

The `FocusTrapZone` registers a global focus handler if its prop `forceFocusInsideTrap` is true and a global click handler unless `isClickableOutsideFocusTrap` is false. However, it only set those handlers once in `componentDidMount` and didn't update them in response to prop changes. This created problems for the scenario of a panel which sets `isHiddenOnDismiss=true` (so it's hidden not destroyed on dismiss): it was impossible to make the panel capture clicks/focus while shown, but go back to not capturing clicks while hidden.

Fix is to add/remove the event handlers as appropriate in both `componentDidMount` and `componentWillReceiveProps`.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/6741)

### CODE REVIEW

**Comment by natalieethell** [comment]
Thanks for adding me! I would suggest adding a comment similar to what I added in Panel.types in the original PR: https://github.com/OfficeDev/office-ui-fabric-react/pull/5805/files#diff-1bd227d6e70f246fc35477b5024fd30fR45

Also, just wondering, is there a reason for not adding this test as well? https://github.com/OfficeDev/office-ui-fabric-react/pull/5805/files#diff-6abe37aec659548639ad07019886e189R39

**Comment by ecraig12345** [comment]
@natalieethell Main reason was "laziness," manually copying the core change rather than cherry-picking...but I can update my branch to include your whole change.

**Comment by natalieethell** [comment]
I think it would be best to either add the test or cherry-pick. :)

File: apps/vr-tests/src/stories/Panel.stories.tsx [hunk 1]
```diff
@@ -3,67 +3,44 @@ import * as React from 'react';
...
+
 storiesOf('Panel', module)
   .addDecorator(FabricDecorator)
   .addDecorator(story => (
     <Screener
-      steps={ new Screener.Steps()
+      steps={new Screener.Steps()
+        .snapshot('default')
+        .click('.ms-SearchBox-field')
         .snapshot('default')
```
[the remark this answers is not in the document]
Comment (natalieethell) [reply apps/vr-tests/src/stories/Panel.stories.tsx]: Have a PR to change that in master here: https://github.com/OfficeDev/office-ui-fabric-react/pull/6764

**Pull request merged**